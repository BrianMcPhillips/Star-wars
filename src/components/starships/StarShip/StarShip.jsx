import React from 'react';
import PropTypes from 'prop-types';

const StarShip = ({ name, model, url }) => (
  <>
    <h1>{name}</h1>
    <h3>{model}</h3>
  </>
);

StarShip.propTypes = {
  name: PropTypes.string.isRequired,
  model: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default StarShip;
