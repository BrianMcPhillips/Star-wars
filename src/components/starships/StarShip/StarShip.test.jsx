import React from 'react';
import { render, cleanup } from '@testing-library/react';
import StarShip from './StarShip';

describe('StarShip component', () => {
  afterEach(() => cleanup());
  it('renders StarShip', () => {
    const { asFragment } = render(
      <StarShip 
        name="Star Destroyer"
        model="Imperial I-class Star Destroyer"
        url="https://swapi.dev/api/starships/3/"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
