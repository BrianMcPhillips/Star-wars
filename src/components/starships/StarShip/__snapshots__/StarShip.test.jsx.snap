// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarShip component renders StarShip 1`] = `
<DocumentFragment>
  <h1>
    Star Destroyer
  </h1>
  <h3>
    Imperial I-class Star Destroyer
  </h3>
</DocumentFragment>
`;
